actor-profile.js

```diff
@@ -0,0 +1,96 @@
+import { db } from "./dbInstance.js";
+import { fetchActorInfo } from "./post.js";
+
+class ActorProfile extends HTMLElement {
+  static get observedAttributes() {
+    return ["url"];
+  }
+
+  constructor() {
+    super();
+    this.url = "";
+  }
+
+  connectedCallback() {
+    this.url = this.getAttribute("url");
+    this.fetchAndRenderActorProfile(this.url);
+  }
+
+  async fetchAndRenderActorProfile(url) {
+    const actorInfo = await fetchActorInfo(url);
+    console.log(actorInfo);
+    if (actorInfo) {
+      this.renderActorProfile(actorInfo);
+      this.updateFollowButtonState();
+      // Update distributed-outbox URL based on fetched actorInfo
+      const distributedOutbox = document.querySelector("distributed-outbox");
+      distributedOutbox.setAttribute("url", actorInfo.outbox);
+    }
+  }
+
+  renderActorProfile(actorInfo) {
+    // Clear existing content
+    this.innerHTML = "";
+
+    const profileContainer = document.createElement("div");
+    profileContainer.classList.add("actor-profile");
+
+    // Create a container for the actor icon and name, to center them
+    const actorContainer = document.createElement("div");
+    actorContainer.classList.add("actor-container");
+
+    // Handle both single icon object and array of icons
+    let iconUrl = './assets/profile.png'; // Default profile image path
+    if (actorInfo.icon) {
+      if (Array.isArray(actorInfo.icon) && actorInfo.icon.length > 0) {
+        iconUrl = actorInfo.icon[0].url;
+      } else if (actorInfo.icon.url) {
+        iconUrl = actorInfo.icon.url;
+      }
+    }
+  
+    const img = document.createElement("img");
+    img.classList.add("actor-icon");
+    img.src = iconUrl;
+    img.alt = actorInfo.name ? actorInfo.name : "Actor icon";
+    actorContainer.appendChild(img); // Append to the actor container
+
+    if (actorInfo.name) {
+      const pName = document.createElement("div");
+      pName.classList.add("actor-name");
+      pName.textContent = actorInfo.name;
+      actorContainer.appendChild(pName); // Append to the actor container
+    }
+
+    // Append the actor container to the profile container
+    profileContainer.appendChild(actorContainer);
+
+    // Create and position the follow button
+    const followButton = document.createElement("button");
```

Comment:
Maybe this could be a seperate component that stores all this state? The styles could also target it. It could have a `state` attribute which it will update and you can target with css like `follow-button[state=whatever]`, and a `url` field it can use to fetch its state from the db